pelican/tools/pelican_import.py

```diff
@@ -536,6 +544,131 @@ def tumblr2fields(api_key, blogname):
         posts = _get_tumblr_posts(api_key, blogname, offset)
 
 
+def strip_medium_post_content(soup) -> str:
+    """Strip some tags and attributes from medium post content.
+
+    For example, the 'section' and 'div' tags cause trouble while rendering.
+
+    The problem with these tags is you can get a section divider (--------------)
+    that is not between two pieces of content.  For example:
+
+      Some text.
+
+      .. container:: section-divider
+
+         --------------
+
+      .. container:: section-content
+
+      More content.
+
+    In this case, pandoc complains: "Unexpected section title or transition."
+
+    Also, the "id" and "name" attributes in tags cause similar problems.  They show
+    up in .rst as extra junk that separates transitions.
+    """
+    # Remove tags
+    # section and div cause problems
+    # footer also can cause problems, and has nothing we want to keep
+    # See https://stackoverflow.com/a/8439761
+    invalid_tags = ["section", "div", "footer"]
+    for tag in invalid_tags:
+        for match in soup.findAll(tag):
+            match.replaceWithChildren()
+
+    # Remove attributes
+    # See https://stackoverflow.com/a/9045719
+    invalid_attributes = ["name", "id", "class"]
+    bs4 = _import_bs4()
+    for tag in soup.descendants:
+        if isinstance(tag, bs4.element.Tag):
+            tag.attrs = {
+                key: value
+                for key, value in tag.attrs.items()
+                if key not in invalid_attributes
+            }
+
+    # Get the string of all content, keeping other tags
+    all_content = "".join(str(element) for element in soup.contents)
+    return all_content
+
+
+def mediumpost2fields(filepath: str) -> tuple:
+    """Take an HTML post from a medium export, return Pelican fields."""
+
+    soup = file_to_soup(filepath, "html.parser")
+    assert soup, f"{filepath} could not be parsed by beautifulsoup"
+    kind = "article"
+
+    content = soup.find("section", class_="e-content")
+    assert content, f"{filepath}: Post has no content"
```

Comment:
Please don't use `assert` for flow control.

```python
if condition:
    raise RuntimeError(...)  # or if possible a more relevant exception
```

Also, you don't handle this error in `mediumposts2fields`. So if a single file triggers these, whole process is halted. Is that the desired behavior or is it better to warn about problematic ones and continue with what it can process?

Reply:
It might be better to continue, but that's not implemented here.

Reply:
Changed two asserts to raise exceptions.

In general, I use asserts when I expect something to never happen, but I don't remember here exactly because it's been too long.